### PR TITLE
bugfix - Regression in transaction_add API blocking saving transaction to Buxfer

### PR DESCRIPTION
**Pull request by nisan-tagar**
A regression in Buxfer transaction_add API was found and reported to support, blocking the ability to add new transactions.

A workaround was suggested by support, and I implemented it for the Buxfer ETL storage code: **keeping the transaction status undefined.**

When the issue is resolved, I will revert this change to re-populate the status

### CODE REVIEW

File: src/storage/buxfer.ts [hunk 1]
```diff
@@ -117,10 +117,19 @@ export class BuxferStorage implements TransactionStorage {
...
+
+  /**
+   * Buxfer does not allow updating all trx fields via REST API so this will add additional fields to the description with '|' separators
+   * @param tx Converted transaction from scrapper
+   * @returns Composite description string
+   */
+  private createCompositeDescription(tx: TransactionRow): string {
+    const compositeDescription: string = `${tx.description}${tx.memo != null && tx.memo.length > 0 ? ` | ${tx.memo}` : ""}`;
+    return compositeDescription;
+  }
```
Comment (daniel-hauser) [inline src/storage/buxfer.ts]:
```suggestion
  private createCompositeDescription(tx: TransactionRow): string {
    return [tx.description, tx.memo].filter(Boolean).join(" | ");
  }
```

File: src/storage/buxfer.ts [hunk 2]
```diff
@@ -117,10 +117,19 @@ export class BuxferStorage implements TransactionStorage {
       accountId: Number(accountId),
       date: format(parseISO(tx.date), BUXFER_DATE_FORMAT, {}),
       amount: tx.chargedAmount,
-      description: tx.description,
-      status:
-        tx.status === TransactionStatuses.Completed ? "cleared" : "pending",
+      description: this.createCompositeDescription(tx),
+      // TODO - uncomment once the API bug is resolved on Buxfer service --> status: = tx.status === TransactionStatuses.Completed ? "cleared" : "pending",
```
Comment (daniel-hauser) [inline src/storage/buxfer.ts]: Can you remove the field in the `buxfer-ts-client` code? Then when they fix it the only change we need in moneyman is the version

```suggestion
      description: this.createCompositeDescription(tx),
      status: tx.status === TransactionStatuses.Completed ? "cleared" : "pending",
```

File: src/storage/buxfer.ts [hunk 3]
```diff
@@ -117,10 +117,19 @@ export class BuxferStorage implements TransactionStorage {
       accountId: Number(accountId),
       date: format(parseISO(tx.date), BUXFER_DATE_FORMAT, {}),
       amount: tx.chargedAmount,
-      description: tx.description,
-      status:
-        tx.status === TransactionStatuses.Completed ? "cleared" : "pending",
+      description: this.createCompositeDescription(tx),
```
Comment (daniel-hauser) [inline src/storage/buxfer.ts]: Considre inlining
```suggestion
      description: [tx.description, tx.memo].filter(Boolean).join(" | "),
```

File: src/storage/buxfer.ts (hunk 1 above)
Comment (daniel-hauser) [reply src/storage/buxfer.ts]: See also https://github.com/daniel-hauser/moneyman/pull/366#discussion_r1823420562

File: src/storage/buxfer.ts (hunk 3 above)
Comment (nisan-tagar) [reply src/storage/buxfer.ts]: Thanks for this

File: src/storage/buxfer.ts (hunk 1 above)
Comment (nisan-tagar) [reply src/storage/buxfer.ts]: Thanks! I accepted #366

File: src/storage/buxfer.ts (hunk 2 above)
Comment (nisan-tagar) [reply src/storage/buxfer.ts]: Got it, will do.

Comment (nisan-tagar) [reply src/storage/buxfer.ts]: Got it, done.

**Comment by nisan-tagar** [comment]
Thanks @daniel-hauser for the great remarks, I accepted them all such that when the regression is fixed on Buxfer API it would only require a client bump to resolve it here as well.

**Pull request merged**